### ISSUE TITLE
Failed to validation package signing for 3rd-party workload packs

### ISSUE THREAD

**Issue by WonyoungChoi** [dotnet/sdk#20381]
I wonder if there are any policies regarding installing 3rd-party workloads in the dotnet 6.0 sdk. 

I am trying to make a third-party workload for providing `net6.0-tizen` TFM in .NET 6.0. This is very important factor to support the `tizen` backend of MAUI. 
- Tizen workload repository : https://github.com/Samsung/Tizen.NET/tree/main/workload
- Tizen workload manifest package: https://www.nuget.org/packages/Samsung.NET.Sdk.Tizen.Manifest-6.0.100/

However, If I create the workload manifest files in `<dotnet>/sdk-manifests/6.0.100/samsung.net.sdk.tizen` directory and try to install workload packs using `dotnet workload install tizen`, the following error occurs.
```html
Installing pack Samsung.Tizen.Sdk version 6.5.100-rc.1.92...
Workload installation failed, rolling back installed packs...
Rolling back pack Samsung.Tizen.Sdk installation...
Workload installation failed: Failed to validate package signing.

Verifying Samsung.Tizen.Sdk.6.5.100-rc.1.92

Signature type: Repository
  Subject Name: CN=NuGet.org Repository by Microsoft, O=NuGet.org Repository by Microsoft, L=Redmond, S=Washington, C=US
  SHA256 hash: 5A2901D6ADA3D18260B9C6DFE2133C95D74B9EEF6AE0E5DC334C8454D1477DF4
  Valid from: 2/16/2021 9:00:00 AM to 5/16/2024 8:59:59 AM
``` 

In the following code, I found that only first-party packages are allowed to be downloaded in a Windows environment.
https://github.com/dotnet/sdk/blob/d01e5c7faad6b27a7d814d27ed0ac9ab0699cb63/src/Cli/dotnet/NugetPackageDownloader/FirstPartyNuGetPackageSigningVerifier.cs#L46

After all, third party workloads cannot be installed with the workload install command, is this the intended policy?
Of course, it is possible to avoid the download process by downloading all the workload packs in advance. But I want to know the exact policy of the dotnet workload.

**Comment by AraHaan** [comment]
I agree that people should be allowed to install 3rd party workloads on Windows as well if they can do it on linux and macos.

**Comment by rookiejava** [comment]
Any update on this?

**Comment by AraHaan** [comment]
Currently not really but I sorta worked around this limitation by making dotnet/arcade installers for dev builds and my own .NET crossplatform global tool that basically installs what is not installed manually by the installers using nuget.org published packages and basically creates the workload manifest information itself. After some careful trial and errors I did make it work perfectly as well as the maui workload 😼.

**Comment by marcpopMSFT** [comment]
At this time, there is not support for 3rd party workloads as we don't have a way of checking the validity of those workloads. @Redth I believe you were engaged on Tizen on how to enable this scenario.

**Comment by AraHaan** [comment]
I guess 1 option is to check if they are using a valid digital signature that could be issued to them by Microsoft. That way the only way they can install an 3rd party workload is if Microsoft in some way checks the code to be malicious free (via automation of some sorts).

However doing that might be tricky unless they be forced to create an account, get an API key, then submit their workload nuget packages to it for signing, and then after that they would have to download the signed packages and then push those signed packages to nuget.org.

**Comment by joeloff** [comment]
It's not so much about being signed, though that's part of the issue. For admin based installations, because they run elevated, installers that contain custom actions can effectively execute arbitrary code at install time.
